app/controllers/candidate_interface/additional_referees_controller.rb

```diff
@@ -0,0 +1,93 @@
+module CandidateInterface
+  class AdditionalRefereesController < CandidateInterfaceController
+    rescue_from ActiveRecord::RecordNotFound, with: :render_404
+
+    def new
+      redirect_to_confirm_if_no_more_reference_needed
+      @reference = ApplicationReference.new
+    end
+
+    def create
+      redirect_to_confirm_if_no_more_reference_needed
+      reference = current_application.application_references.build(referee_params)
+
+      if reference.save
+        redirect_to_confirm_or_show_another_reference_form
+      else
+        @reference = reference
+        render :new
+      end
+    end
+
+    def confirm
+      redirect_to_dashboard_if_no_references_needed
+      @references = find_references
+    end
+
+    def request_references
+      references_to_confirm = find_references.includes(:application_form).to_a
+
+      references_to_confirm.each do |reference|
+        RefereeMailer.reference_request_email(current_candidate.current_application, reference).deliver_now
+        reference.update!(feedback_status: 'feedback_requested')
+      end
+
+      flash[:success] = I18n.t!('additional_referees.feedback_flash', count: references_to_confirm.size)
+
+      redirect_to candidate_interface_application_form_path
+    end
+
+    def edit
+      @reference = find_reference
+    end
+
+    def update
+      reference = find_reference
+
+      if reference.update(referee_params)
+        redirect_to_confirm_or_show_another_reference_form
+      else
+        @reference = reference
+        render :edit
+      end
+    end
+
+  private
+
+    def find_reference
```

Comment:
```suggestion
    def current_reference
```

```suggestion
    def reference
```

What do you think of either of these names?